src/rust-crypto/RoomEncryptor.ts

```diff
@@ -121,16 +122,20 @@ export class RoomEncryptor {
      * in the room.
      *
      * @param globalBlacklistUnverifiedDevices - When `true`, it will not send encrypted messages to unverified devices
+     * @param deviceIsolationMode - The device isolation mode see {@link DeviceIsolationMode}
      */
-    public async prepareForEncryption(globalBlacklistUnverifiedDevices: boolean): Promise<void> {
+    public async prepareForEncryption(
+        globalBlacklistUnverifiedDevices: boolean,
+        deviceIsolationMode: DeviceIsolationMode = new AllDevicesIsolationMode(false),
```

Comment:
given that this is only called in one place (except in tests), we should not provide a default value.

Likewise below.

Reply:
https://github.com/matrix-org/matrix-js-sdk/pull/4425/commits/670fe2e5d84dcc120cc280ee98b1cc0ae024592d